src/system/Zikula/Module/SearchModule/AbstractSearchable.php

```diff
@@ -13,6 +13,7 @@
 
 namespace Zikula\Module\SearchModule;
 
+use Symfony\Component\DependencyInjection\ContainerInterface;
```

Comment:
The SearchModule would define a service called `zikulasearchmodule.searchable_chain` or something. This service will hold the list of all the searchable modules. To do so, every module who wants to use the new search stuff would need to create a service themself, with a tag like `<tag name="zikula.searchable" />` in it's service definition. The modules' services would need to implement a SearchableInterface which defines the needed methods (like `getResults`).
Then, in a compiler pass of the SearchModule, you would do what's done here http://symfony.com/doc/current/components/dependency_injection/tags.html#create-a-compilerpass.

Then, when a search is issued, the SearchModule can use it's own service, which now has a list of all the searchable services, and get the results one by one.

The advantage is that literally every bundle (not only modules) can provide a search. Also, the bundles don't need to extend some abstract class.